orchagent/portsorch.cpp

```diff
@@ -1703,6 +1696,13 @@ void PortsOrch::doPortTask(Consumer &consumer)
                 m_pendingPortSet.erase(alias);
             }
 
+            if (!m_portConfigDone)
+            {
+                // Not yet receive PortConfigDone. Save it for future retry
+                it++;
+                continue;
+            }
```

Comment:
Azure/sonic-mgmt#834 (comment) is a good analysis, which explains why you move the section to here